apps/extension/src/routes/popup/home/index.tsx

```diff
@@ -33,32 +38,31 @@ export const popupIndexLoader = async (): Promise<Response | PopupLoaderData> =>
 };
 
 export const PopupIndex = () => {
-  const account = useStore(selectedAccount);
-  const { next, previous, setIndex } = useStore(accountsSelector);
+  const active = useStore(getActiveWallet);
+
+  const getAccount = (index: number, ephemeral: boolean) => {
+    if (!active) return;
+
+    const addr = ephemeral
+      ? getEphemeralByIndex(active.fullViewingKey, index)
+      : getAddressByIndex(active.fullViewingKey, index);
+    const bech32Addr = bech32Address(addr);
+
+    return {
+      address: bech32Addr,
+      preview: ephemeral
+        ? bech32Addr.slice(0, 33) + '…'
+        : getShortAddressByIndex(active.fullViewingKey, index),
+      index,
+    };
+  };
```

Comment:
This logic should be a selector function that lives inside /state/wallets.

```typescript
const getAccount = useStore(accountAddrSelector);
```

Reply:
do I need to store the value of the function (selectedAccount) in /state/wallets?

Reply:
what do you mean?

Reply:
i add fucntion to state.wallets which return account address:
```
getAccountAddr: (index: number, ephemeral: boolean) => {
        const active = getActiveWallet(get());
        if (!active) return;

        const addr = ephemeral
          ? getEphemeralByIndex(active.fullViewingKey, index)
          : getAddressByIndex(active.fullViewingKey, index);
        const bech32Addr = bech32Address(addr);

        return {
          address: bech32Addr,
          preview: ephemeral
            ? bech32Addr.slice(0, 33) + '…'
            : getShortAddressByIndex(active.fullViewingKey, index),
          index,
        };
      },
```
Should i save this addresses to WalletsSlice? whether the user will be able to select the account index and when closing/opening the extension, he will be see the one he specified index, not 0 index

Reply:
No, not within the slice. It's a selector:

```typescript
export const walletsSelector = (state: AllSlices) => state.wallets;
export const getActiveWallet = (state: AllSlices) => state.wallets.all[0];

export const accountAddrSelector = (state: AllSlices) => (index: number, ephemeral: boolean) => {
  const active = getActiveWallet(state);
  if (!active) return;

  const addr = ephemeral
    ? getEphemeralByIndex(active.fullViewingKey, index)
    : getAddressByIndex(active.fullViewingKey, index);
  const bech32Addr = bech32Address(addr);

  return {
    address: bech32Addr,
    preview: ephemeral
      ? bech32Addr.slice(0, 33) + '…'
      : getShortAddressByIndex(active.fullViewingKey, index),
    index,
  };
};

=======

export const PopupIndex = () => {
  const getAccount = useStore(accountAddrSelector);

  return (
    <div className='relative flex h-full flex-col items-stretch justify-start bg-left-bottom px-[30px]'>
      <div className='absolute bottom-[50px] left-[-10px] -z-10 h-[715px] w-[900px] overflow-hidden bg-logo opacity-10' />
      <IndexHeader />
      <div className='my-32'>
        <SelectAccount getAccount={getAccount} />
      </div>
      <BlockSync />
    </div>
  );
};


```

Reply:
ok, thanks